dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
+  
+  for(var i = 0; i < cookie.length; i++) {
+    var pair = cookie[i].split('=');
+    var key = pair[0].trim();
+
+    switch(key) {
+      case 'progress':
+        var allLevelsProgress = JSON.parse(decodeURIComponent(pair[1]));
+        for(var level in allLevelsProgress) {
+          if (allLevelsProgress.hasOwnProperty(level)) {
```

Comment:
One nit that's somewhat personal preference, you could reduce nesting here by saying
```
if (!allLevelsProgress.hasOwnProperty(level)) {
  break;
}
...
```

Also, because this is an object we got form JSON, I think the hasOwnProperty check might be superfluous (will always return true).

You could also do this using forEach, which is what I personally tend to do, but again that's largely personal preference and either approach is fine.
```
allLevelsProgress.forEach(function (level) {
 ...
});
```